jre_emul/Classes/com/google/j2objc/security/IosRSAKeyPairGenerator.java

```diff
@@ -74,10 +85,31 @@ public native KeyPair generateKeyPair() /*-[
     JavaSecurityKeyPair *keyPair =
         AUTORELEASE([[JavaSecurityKeyPair alloc] initWithJavaSecurityPublicKey:publicKey
                                                     withJavaSecurityPrivateKey:privateKey]);
+
     [publicKey release];
     [privateKey release];
     return keyPair;
   ]-*/;
+  
+  /*-[
+  -(BOOL) deleteKey:(NSString *)tag   
+  		   keyClass:(CFStringRef) keyClass {
+  		   
+    NSData *publicTag = [tag dataUsingEncoding:NSUTF8StringEncoding];
+
+    NSMutableDictionary *query = [NSMutableDictionary dictionary];
+    query[(id)kSecClass] = (id)kSecClassKey;
+    query[(id)kSecAttrKeyType] = (id)kSecAttrKeyTypeRSA;
+    query[(id)kSecAttrKeyClass] = (id)keyClass;
+    query[(id)kSecAttrApplicationTag] = tag;
+	OSStatus status2 = SecItemDelete((CFDictionaryRef) query);
+    if (status2 != errSecSuccess && status2 != errSecItemNotFound) {
+        NSLog (@"Problem removing previous public key from the keychain, OSStatus == %d", (int)status2);
+        return false;
+    }
```

Comment:
I tried to throw an exception but got the following error: 
@throw is causing this error 
Implicit declaration of function 'create_JavaSecurityProviderException_initWithNSString_' is invalid in C99  [-Werror,-Wimplicit-function-declaration]

Don't know why.